modules/parsers/carousel.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Get the common ancestor of two or more elements
+ * @param {NodeList} elements The list of elements to compare
+ * @returns {Node} The common ancestor
+ */
+function getCommonAncestor(elements) {
+  // If there are no elements, return null
+  if (elements.length === 0) {
+    return null;
+  }
+
+  // If there's only one element, return it
+  if (elements.length === 1) {
+    return elements[0];
+  }
+
+  // Otherwise, create a new Range
+  const range = document.createRange();
+
+  // Start at the beginning of the first element
+  range.setStart(elements[0], 0);
+
+  // Stop at the end of the last element
+  range.setEnd(
+    elements[elements.length - 1],
+    elements[elements.length - 1].childNodes.length,
+  );
+
+  // Return the common ancestor
+  return range.commonAncestorContainer;
+}
+
+export default function parse(el) {
+  // a carousel will consist of one row for every image found
+  const images = el.querySelectorAll('img');
+  const commonParent = getCommonAncestor(images);
+  return [...images].map((img) => {
+    const slide = [...commonParent.children].find((child) => child.contains(img));
+    const content = [...slide.children].filter((child) => !child.contains(img));
+    return [img, content];
+  });
```

Comment:
I'm sure this pattern won't work for all carousels out there but its a start.

Reply:
It is a good start- covering the 'img' case.  I've seen background-image used instead of 'img' - that could be tricky.  So many ways to do the same thing.

Reply:
background images would be tougher since that would require parsing CSS. Current parsing relies on content already existing in the DOM.